src/core/inc/bao.h

```diff
@@ -14,15 +14,26 @@
 #include <console.h>
 #include <util.h>
 
-#define INFO(args, ...)    console_printk("BAO INFO: " args "\n" __VA_OPT__(, ) __VA_ARGS__);
 
-#define WARNING(args, ...) console_printk("BAO WARNING: " args "\n" __VA_OPT__(, ) __VA_ARGS__);
+#define INFO(...)                                         \
+    do {                                                   \
+        console_printk("BAO INFO: " __VA_ARGS__); \
+        while (true) { }                                   \
+    } while (true);
```

Comment:
I think what you want here is a single iteration while loop so it would be better to:

```suggestion
    do {                                                   \
        console_printk("BAO INFO: " __VA_ARGS__); \
        while (true) { }                                   \
    } while (false)
```

Reply:
You also frame the commit message from this commit regarding only the VA_OPT issue. No mention why you introduce the `do { ... } while()` loops.

Reply:
Fixed in commit e6bc02bdc72f79dcdb517828a42d7e57842c3ed5